File: SPOJ/11375.py
t=input()
for i in range(1,t+1):
	x=raw_input().split()
	n=int(x[0])
	m=int(x[1])
	r1 = []
	r2 = []
	spy = 0
	while m:
		m = m - 1
		y=raw_input().split()
		if y[0] in r2:
			spy = 1
		else:
			r1.append(y[0])
		if y[1] in r1:
			spy = 1
		else:
			r2.append(y[1])
	print "Scenario #"+str(i)+":",
	if spy==1:
		print "spied"
	else:
		print "spying"

File: SPOJ/11375.c
#include<stdio.h>
int main()
{
	int test_cases,t;
	scanf("%d",&test_cases);
	for(t=1;t<=test_cases;t++)
	{
		int m,r1,r2,spy = 0;
		int a[100000]={0},b[100000]={0};
		scanf("%*d%d",&m);
		while(m--)
		{
			scanf("%d%d",&r1,&r2);
			if(b[r1] == 1)
				spy=1;
			else
				a[r1]=1;
			if(a[r2] == 1)
				spy=1;
			else
				b[r2]=1;
		}
		printf("Scenario #%d: %s\n",t,(spy==0)?"spying":"spied");
	}
	return 0;
}
